### PR TITLE
AliExternalTrackParam4D::PropagateToMirrorX for looper simulation and fits

### PR DESCRIPTION
**Pull request by Trenchcoat95**
Created a PropagateToMirrorX function to propagate "loopers" correctly and made changes to other functions accordingly

### CODE REVIEW

**Comment by miranov25** [comment]
## propagateToMirrorX functions

* assumes  no MB correction
* figure with explanation needed
  * left side /right side

### Mirror transformation:
* qPt,sin(phi),tgl - changing sign
* alpha is mirrored in respect to symmetry plan
* local Y and local Z - propagated along arc assuming constant curvature

File: fastMCKalman/MC/fastSimulation.cxx [hunk 1]
```diff
@@ -215,7 +294,7 @@ Bool_t AliExternalTrackParam4D::GetXYZatR(Double_t xr,Double_t bz, Double_t *xyz
 }
 /// get radiial direction sign
 Int_t AliExternalTrackParam4D::GetDirectionSign(){
-  Float_t dir = GetX()*Px()+GetY()*Py();
+  Float_t dir = GetX()*Px()+GetY()*Py()+GetZ()*Pz();
```
Comment (miranov25) [inline fastMCKalman/MC/fastSimulation.cxx:297]: Should be reverted- this function should check of the moneta is out or in in cylindrical coordinate

**Comment by miranov25** [comment]
### Index fix - to be described

**Comment by miranov25** [comment]
### The Z>1 particle removed - to be reintroduced later

File: fastMCKalman/MC/fastSimulation.cxx (hunk 1 above)
Comment (Trenchcoat95) [reply fastMCKalman/MC/fastSimulation.cxx:297]: Got the function back to how it was

**Comment by miranov25** [comment]
The code not yet fully working but be merged to master, we will fix it there.

- problem in the ReconstructParticleFull
- normal reconstruct is working

**Comment by miranov25** [comment]
Relates to #9

**Pull request merged**